shorty.go

```diff
@@ -14,18 +14,49 @@ type Shorty struct {
 
 // New returns a new shorty, machineID can be any unique id to embed in the generated ids.
 func New(store Store, machineID uint32) *Shorty {
-	return &Shorty{
+	s := &Shorty{
 		s:         store,
 		machineID: machineID,
 	}
+	go s.cleanup()
+	return s
+}
+
+func (s *Shorty) cleanup() {
+	for {
```

Comment:
Also, might be good to have the actual clean func exposed so it could be called manually (for whatever reason)
^^^ This isn't a requirement, just a suggestion